ltm/agent.py

```diff
@@ -259,56 +283,39 @@ def keywords_for_message(self, user_message, cost_cb):
         print(f"Interaction keywords: {keywords}")
         return keywords
 
-    def create_context(self, user_message, max_prompt_size, previous_interactions, cost_cb):
-        context = [make_system_message(self.system_message), make_user_message(
-            f"{str(self.now)[:-7]} ({td_format(datetime.timedelta(seconds=1))}) " + user_message)]
-        relevant_interactions = self.get_relevant_memories(user_message, cost_cb)
-
-        # Get interactions from the memories
-        for m in relevant_interactions:
-            if "trivia" in m[0].content:
-                colour_print("YELLOW", f"<*** trivia ***>")
-            else:
-                colour_print("YELLOW", f"{m[0].content}")
+    def collect_memories(self, user_message: str, previous_interactions: int) -> LLMContext:
+        relevant_interactions = self.get_relevant_memories(user_message)
```

Comment:
When counting tokens, this function does not take the size of the current user message into account.

Reply:
Ah, yes it does. I created a property `max_input_size`, which is `max_prompt_size - max_message_size`.

Reply:
So we need to sort out our nomenclature here. From the top the LTMAgentWrapper has `max_prompt_size` and `max_message_size`. 

- First off: What message does `max_message_size` refer to? The input or output?

In `LTMAgentWrapper` instantiates `LTMReflectionAgent` with `max_message_size -> max_completion_tokens`

- So now its the output message size?

In `LTMAgent` we have `max_input_tokens` which is  `0.9 * (max_prompt_size - completion_size)`, so that takes account of the output tokens of the model, OR the input size, but not both.

There is also a `max_message_size` at line 66 which is never used.

Reply:
Right, I agree on the `max_message_size` issue. I think it should apply only to input messages, and we should have a much shorter length for completion. I applied the `0.9` factor in `max_input_tokens` because there were slight errors in early versions and I didn't want to deal with them, but we can hopefully get rid of it now. `max_message_size` was originally established to communicate the scheduler the agent's capability for taking inputs, so what about defining a `max_response_size` for the output?